plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadataConfig.java

```diff
@@ -100,6 +102,11 @@ public int getDomainCompactionThreshold()
         return domainCompactionThreshold;
     }
 
+    public int getColumnAliasMaxChars()
```

Comment:
This is not needed. This shouldn't be user configurable property but something that is exposed by the implementations extending JdbcClient. Move this information to the JdbcClient with the default implementation returning `OptionalInt.empty()` (meaning no identifier length limit)

Reply:
@wendigo, I don't think you got to the clue of the problem. You have provided me with the test case example here:
https://github.com/trinodb/trino/pull/18858/files#diff-dcd58a1db0b844e3264313bafcbdd64e74bdfc8b6c864aa776fe361f9c13888e

The exact code from what you provided, I'm referring to is:
```
    @Test
    public void testJoinPushdownWithLongIdentifiers()
    {
        String maximumColumnIdentifier = "a".repeat(ORACLE_MAXIMUM_IDENTIFIER_LENGTH);

        // Verify that the ORACLE_MAXIMUM_IDENTIFIER_LENGTH is correct for the Oracle version
        // used in the test.
        oracleServer.execute("SELECT 1 AS " + maximumColumnIdentifier + " FROM DUAL");
        assertThatThrownBy(() -> {
            oracleServer.execute("SELECT 1 AS " + maximumColumnIdentifier + "a FROM DUAL");
        }).hasMessageContaining("ORA-00972: identifier is too long");

        try (TestTable table = new TestTable(onRemoteDatabase(), "join_long_", "(%s decimal(19, 0))".formatted(maximumColumnIdentifier))) {
            assertThat(query(joinPushdownEnabled(getSession()), """
                SELECT c.custkey, o.%s, n.nationkey
                FROM %s o JOIN customer c ON c.custkey = o.%s
                JOIN nation n ON c.nationkey = n.nationkey""".formatted(maximumColumnIdentifier, table.getName(), maximumColumnIdentifier)))
            .isFullyPushedDown();
        }
    }
```

This code is the equivalent of the test that I already have, which is in: `TestOracleConnectorTest#testPushdownJoinWithLongNameSucceedsWhenConfiguredCorrectly` and is the 'happy path' testing.

I didn't need to introduce the config property nor the method `getColumnAliasMaxChars` in order to code that test, ie. test the happy path. So whatever time you devoted to providing the code sample above, you could have used elsewhere in the project. Please don't take it personally. I'm only writing the above as on our meeting with @hubertsk you have said explicitly that you don't have time to leave detailed justification against the comments you are making in the PR. As your time as a technical leader is valuable, by saying the above, I wanted to give you clues where you can save your precious time. Maybe you wasted it because I somehow mislead you with what kind of help I needed. If that was the case, I apologize.

To go back to the technical part of this discussion. The real help that I'm asking for is how to code that test `TestOracleConnectorTest#testPushdownJoinWithLongNameFailsWhenMaxAllowedCharsIsMisconfigured` (ie. a non-happy path) without adding the config property nor `getColumnAliasMaxChars` method (like you requested). If you could provide the equivalent in the branch you have provided your code samples already, that would be awesome. I am otherwise currently stuck on exactly that issue.

Reply:
I don't think there is a need to provide a negative test here. As I understand positive test (happy path) is good enough here to provide the regression coverage. Notice that test should fail without product code changes you provide. We don't need to prove that we are able to break our code.

Can you please explain me what is the point of negative test here? What do I miss?

> IsMisconfigured

The best way to avoid misconfiguration is disallow users to configure. Which was requested with https://github.com/trinodb/trino/pull/18642#discussion_r1308462728

Reply:
@kokosing  - thank you. Reverted the change with adding a config property and removed the negative test.